extensions/authentication-jwt/README.md

```diff
@@ -0,0 +1,317 @@
+    this.static('/', path.join(__dirname, '../public'));
+
+    // - enable jwt auth -
+    // Add security spec (Future work: refactor it to an enhancer)
+    this.addSecuritySpec();
+    // Mount authentication system
+    this.component(AuthenticationComponent);
+    // Mount jwt component
+    this.component(JWTAuthenticationComponent);
+    // Bind datasource
+    this.dataSource(DbDataSource, UserServiceBindings.DATASOURCE_NAME);
+
+    this.component(RestExplorerComponent);
+    this.projectRoot = __dirname;
+    // Customize @loopback/boot Booter Conventions here
+    this.bootOptions = {};
+  }
+
+  // - enable jwt auth -
+  // Currently there is an extra function to
+  // merge the security spec into the application.
+  // This will be improved with a coming enhancer.
+  // See section [Future Work](#future-work)
+  addSecuritySpec(): void {
+    this.api({
+      openapi: '3.0.0',
+      info: {
+        title: 'test application',
+        version: '1.0.0',
+      },
+      paths: {},
+      components: {securitySchemes: SECURITY_SCHEME_SPEC},
+      security: [
+        {
+          // secure all endpoints with 'jwt'
+          jwt: [],
+        },
+      ],
+      servers: [{url: '/'}],
+    });
+  }
+}
+```
+
+</p>
+</details>
+
+_All the jwt authentication related code are marked with comment "- enable jwt
+auth -", you can search for it to find all the related code you need to enable
+the entire jwt authentication in a LoopBack 4 application._
+
+## Add Endpoint in Controller
+
+After mounting the component, you can call token and user services to perform
+login, then decorate endpoints with `@authentication('jwt')` to inject the
+logged in user's profile.
+
+This module contains an example application in the `fixtures` folder. It has a
+controller with endpoints `/login` and `/whoAmI`.
+
+The code snippet for login function:
+
+```ts
+async login(
+    @requestBody(CredentialsRequestBody) credentials: Credentials,
+  ): Promise<{token: string}> {
+    // ensure the user exists, and the password is correct
+    const user = await this.userService.verifyCredentials(credentials);
+
+    // convert a User object into a UserProfile object (reduced set of properties)
+    const userProfile = this.userService.convertToUserProfile(user);
+
+    // create a JSON Web Token based on the user profile
+    const token = await this.jwtService.generateToken(userProfile);
+
+    return {token};
+  }
+```
+
+The code snippet for whoAmI function:
+
+```ts
+@authenticate('jwt')
+  async whoAmI(): Promise<string> {
+    return this.user[securityId];
+  }
+```
+
+The complete file is in
+[user.controller.ts](https://github.com/strongloop/loopback-next/tree/master/extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts)
+
+## Customize Model
+
+1. Create your own user model and repository by `lb4 model` and
+   `lb4 repository`.
+
+2. The user service requires the user model, to provide your own one, you need
```

Comment:
It makes it sound like copying that file is the only way to do it. The instruction should be more generic.

Suggestion: _Create a corresponding custom `UserService` and bind it to `UserServiceBindings.USER_SERVICE`. Take a look at https://github.com/strongloop/loopback-next/blob/master/extensions/authentication-jwt/src/services/user.service.ts for an example of `UserService` implementation._